src/user/user.service.ts

```diff
@@ -1,26 +1,44 @@
 import { Injectable } from '@nestjs/common';
-import { CreateUserDto } from './dto/create-user.dto';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
 import { UpdateUserDto } from './dto/update-user.dto';
+import { User } from './entities/user.entity';
+import * as bcrypt from 'bcrypt';
 
 @Injectable()
 export class UserService {
-  create(createUserDto: CreateUserDto) {
-    return 'This action adds a new user';
-  }
+  constructor(
+    @InjectRepository(User) private userRepository: Repository<User>,
+  ) {}
 
-  findAll() {
-    return `This action returns all user`;
+  findOne(user: User) {
+    return user;
   }
 
-  findOne(id: number) {
-    return `This action returns a #${id} user`;
-  }
+  async update(id: number, updateUserDto: UpdateUserDto) {
+    let user = await this.userRepository.findOneBy({ id });
+
+    const hashSalt = await bcrypt.genSalt();
+
+    user = {
```

Comment:
what if we want to update only one field?

Reply:
UpdateDto has optional properties now so we can change any property of user using Patch